variantlib/utils.py

```diff
@@ -1,8 +1,30 @@
 from __future__ import annotations
 
 from typing import Any
+from typing import TypeVar
 
 
 class classproperty(property):  # noqa: N801
     def __get__(self, cls: Any, owner: type | None = None) -> Any:
         return classmethod(self.fget).__get__(None, owner)()  # type: ignore[arg-type]
+
+
+T = TypeVar("T")
+
+
+def aggregate_user_and_default_lists(
```

Comment:
This can either mutate `user_list` in place or return mutable `default_list`. You should either copy the lists, or at least note the mutability in the docstring, since it can be confusing, especially if `default_list` ends up being a global/shared list somewhere.

Reply:
Oh good point. I don't like mutability. Makes bug tracking very hard to track

Reply:
Fixed